server/src/main/java/org/opensearch/index/codec/composite/Composite99DocValuesReader.java

```diff
@@ -64,24 +207,129 @@ public SortedSetDocValues getSortedSet(FieldInfo field) throws IOException {
     @Override
     public void checkIntegrity() throws IOException {
         delegate.checkIntegrity();
-        // Todo : check integrity of composite index related [star tree] files
+        CodecUtil.checksumEntireFile(dataIn);
     }
 
     @Override
     public void close() throws IOException {
         delegate.close();
-        // Todo: close composite index related files [star tree] files
+        boolean success = false;
+        try {
+            IOUtils.close(metaIn, dataIn);
+            for (DocValuesProducer docValuesProducer : compositeDocValuesProducerMap.values()) {
+                IOUtils.close(docValuesProducer);
+            }
+            success = true;
+        } finally {
+            if (!success) {
+                IOUtils.closeWhileHandlingException(metaIn, dataIn);
+            }
+            starTreeMap.clear();
+            compositeIndexMetadataMap.clear();
+            compositeDocValuesProducerMap.clear();
+            metaIn = null;
+            dataIn = null;
+        }
     }
 
     @Override
     public List<CompositeIndexFieldInfo> getCompositeIndexFields() {
-        // todo : read from file formats and get the field names.
-        return new ArrayList<>();
+        return compositeFieldInfos;
     }
 
     @Override
     public CompositeIndexValues getCompositeIndexValues(CompositeIndexFieldInfo compositeIndexFieldInfo) throws IOException {
-        // TODO : read compositeIndexValues [starTreeValues] from star tree files
-        throw new UnsupportedOperationException();
+
+        switch (compositeIndexFieldInfo.getType()) {
+            case STAR_TREE:
+                // building star tree values
+                CompositeIndexMetadata compositeIndexMetadata = compositeIndexMetadataMap.get(compositeIndexFieldInfo.getField());
```

Comment:
We can move the entire 'StarTreeField' to 'CompositeIndexFieldInfo' as well instead of initializing for every 'getCompositeIndexValues' Call.